### PR TITLE
chore(ci): rename release branch image's tag

### PR DESCRIPTION
**Pull request by mwangggg**
fixes: #39

### CODE REVIEW

**Comment by mwangggg** [comment]
https://github.com/mwangggg/cryostat-grafana-dashboard/actions/runs/6733847443/job/18303438072

**Comment by andrewazores** [comment]
So with this, the pushed branch name is used:
- `main` -> `main` + `latest`
- `cryostat-vX.Y` -> `X.Y.0`

Right?

I think that's a good enough improvement. I might prefer to have `cryostat-vX.Y -> X.Y`. `X.Y.0`, `X.Y.1` would be better to do using git tags somehow.

**Comment by mwangggg** [comment]
https://github.com/mwangggg/cryostat-grafana-dashboard/actions/runs/6736134387/job/18310819993

**Comment by andrewazores** [comment]
Oh, I meant that the `.0` and `.1` tags should not normally be applied. If there is a push to the branch `cryostat-v2.4` then this should result in the image only being tagged as `2.4`, not as `2.4.0` or `2.4.1`. The `2.4.0` and `2.4.1` image tags would ideally be created somehow based on git tags.

I'm imagining a workflow where we cut a release branch like `cryostat-v2.4` off of main. Any push to `cryostat-v2.4` (ex. PRs getting backported) would result in the `2.4` image tag getting bumped up to the most recent image built from that branch. Image tags like `2.4.0` and `2.4.1` would be additionally applied to container images if they are built from commits that have a git tag like `2.4.0` or `2.4.1` on them. So at some point in time, there would be a container image tagged as both `2.4` and `2.4.0`. If another backport goes into the `cryostat-v2.4` branch, then the `2.4` tag would get updated and point to that newer container image, but `2.4.0` would remain on the previous one that was built from the specific commit with that git tag.

I think `git describe --exact-match` sounds like it would be useful for this (`man git-describe`).

**Comment by mwangggg** [comment]
ohhh I thought the period was a comma

**Review by andrewazores — DISMISSED** [review]
LGTM. WDYT @ebaron ?

**Comment by andrewazores** [comment]
If Elliott also agrees on tagging container images using git tags this way then we should do the same or similar across all of our repos that produce container images, too. This makes the release process a little easier/more automated, and keeps it consistent across repos.

**Comment by ebaron** [comment]
I'm a little concerned with using something like `2.4` to represent a development build. I would naively expect that tag to point to whatever the latest 2.4.x release is. Keeping some kind of suffix like `-dev` or `-snapshot` would make this distinction a lot more obvious.

**Comment by andrewazores** [comment]
That's fair. We could have pushes to the branch `cryostat-vX.Y` tag images as `X.Y-dev`, and then create `X.Y.0` and `X.Y.1` image tags using git tags. 

We could leave it at that, or we could add a small snippet of JS that also tags the image as `X.Y` whenever the push corresponds to a git tag `X.Y.Z`. That could go another step further and also add the image tag `X`.

**Review by aali309 — COMMENTED** [review]
LGTM

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,20 +17,36 @@ jobs:
...
+        echo "::set-output name=image-tag-patch::${output:0:3}"
+        echo "::set-output name=image-tag-release::${output:0:1}"
     - name: Build container images and manifest
       id: buildah-build
       uses: redhat-actions/buildah-build@v2
       with:
         image: ${{ env.CRYOSTAT_GRAFANA_IMG }}
         archs: amd64, arm64
-        tags: ${{ github.ref_name }} ${{ github.ref == 'refs/heads/main' && 'latest' || '' }}
+        tags: ${{ steps.check-branch.outputs.image-tag }} ${{ github.ref == 'refs/heads/main' && 'latest' || '' }} ${{ steps.commit-tag.outputs.image-tag }}
```
Comment (andrewazores) [inline .github/workflows/ci.yml:49]: Should the `-patch` and `-release` outputs also be here?

**Review by andrewazores — DISMISSED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -17,20 +17,36 @@ jobs:
...
+      run: |
+        if [[ ${REF_NAME} == cryostat-v* ]]; then
+          echo "image-tag=${REF_NAME:10}-dev" >> $GITHUB_OUTPUT
+        else
+          echo "image-tag=${REF_NAME}" >> $GITHUB_OUTPUT
+        fi
+    - name: Check commit git tag
+      id: check-commit
+      run: |
+        output=$(git describe --tags --exact-match 2>/dev/null || echo -n '')
```
Comment (andrewazores) [inline .github/workflows/ci.yml:39]: So far we have been using git tags like `v2.3.1`, but that should ideally produce image tags `2.3.1`, `2.3`, and `2`. I think just a simple `| sed -e "s/^v//"` should do the trick to strip off the leading `v` if there is one.

```bash
$ echo "v2.3.1" | sed -e "s/^v//"
2.3.1
$ echo "2.3.1" | sed -e "s/^v//"
2.3.1
```

**Review by andrewazores — APPROVED** [review]

**Pull request merged**